Automattic/pocket-casts-ios
# Tracks: Refactor the PlaybackSource

This does some refactoring:
- Renames the `PlaybackSource` to `AnalyticsSourceProvider`
- Adds `AnalyticsSource` enum
- Fills in all the sources in the enum
- Replace the currentSource String with AnalyticsSource
- Updates all the playbackSource's to analyticsSource

## To test
- 🟢 Green CI
- Code View
- Launch the app and tap around to verify the source is being populated correctly in the events

## Checklist
